### DISCUSSION TITLE
Questions re: CostModel entries

### DISCUSSION THREAD

**Discussion by rhyslbw** [CardanoSolutions/ogmios#143] [Q & A] [answered]
Referring to:
https://github.com/CardanoSolutions/ogmios/blob/0acf18ea5c1c3bcb0a8aa93cefff3f831ff60981/clients/TypeScript/packages/schema/src/index.ts#L1250-L1252

1. What is the key and value here?
2. The [Alonzo CDDL](https://github.com/input-output-hk/cardano-ledger/blob/master/eras/alonzo/test-suite/cddl-files/alonzo.cddl#L305) represents this as `cost_model = [ 166*166 int ]`. How does this align with the value type of `Int64`?

**Comment by KtorZ** [comment, marked as answer]
1. This map refers to the name of the Plutus-core primitives and their associate cost in abstract execution units. (see the genesis file for a list of currently existing values and their cost: https://github.com/input-output-hk/cardano-configurations/blob/master/network/mainnet/genesis/alonzo.json#L28-L193). The abstract execution units is then converted to Ada / Lovelace according to the execution prices, also in the protocol parameters (https://github.com/input-output-hk/cardano-configurations/blob/master/network/mainnet/genesis/alonzo.json#L3-L13).

2. The CDDL syntax `n*m type` refers to a number of occurrences in the structure. `n` (resp. `m`) being the lower (resp. upper) limit. So here, it tells us that there are exactly 166 elements in the list. Looking at the serialization code, it's also pretty clear that only the execution costs are actually serialized -- assuming a canonical ordering for deserialization; said differently, labels are omitted in the on-chain representations.
